File: murnana/Unity-TestDrivenDevelopmentByExample/Assets/TestDrivenDevelopmentByExample/Scripts/Runtime/Franc.cs
namespace TDD
{
    public sealed class Franc : Money
    {
        public Franc(int amount, string currency)
            : base (amount: amount, currency: currency)
        {
        }

        public override Money Times(int multiplier)
        {
            return new Money (amount: m_Amount * multiplier, currency: m_Currency);
        }
    }
}

File: miyata/miyata-reading-circle-tdd/src/main/java/money/Franc.java
package money;

public class Franc extends Money {

    private String currency;

    Franc(int amount) {
        this.amount = amount;
        this.currency = "CHF";
    }
    Money times(int multiplier) {
        return new Franc(amount * multiplier);
    }

    @Override
    String currency() {
        return currency;
    }
}
